nio/service/base.py

```diff
@@ -12,6 +12,14 @@
 from nio.util.threading import spawn
 
 
+class BlockException(Exception):
+
+    """ Raised when a block fails to start, includes the block label.
+    """
+    def __init__(self, *args, label=None):
+        super().__init__(*args)
+        self.label = label
```

Comment:
I know I set this exception class structure up initially but after reviewing it I think it makes more sense to have the block object in this exception rather than just the label. That will let exception handlers do whatever they need to with the block. 
```python
class BlockException(Exception):
    def __init__(self, *args, block=None, **kwargs):
        super().__init__(*args, **kwargs)
        self.block = block
...
except BlockException as be:
    print(be.block.label())
```

Reply:
The block object contains at least one un-picklable type `_thread.RLock`, we can find a way around that if we want the block to be available in `ServiceProcess._ipc_default_handler`. This might be useful for getting the configuration or something from the block.